tests/unit/behavior_flag_tests/test_empty_project.py

```diff
@@ -0,0 +1,87 @@
+from types import SimpleNamespace
+from typing import Any, Dict, List
+
+from dbt_common.behavior_flags import BehaviorFlag
+from dbt_common.exceptions import DbtBaseException
+import pytest
+
+from dbt.adapters.contracts.connection import AdapterRequiredConfig, QueryComment
+
+from tests.unit.fixtures.credentials import CredentialsStub
+
+
+@pytest.fixture
+def flags() -> Dict[str, Any]:
+    return {
+        "unregistered_flag": True,
+        "default_false_user_false_flag": False,
+        "default_false_user_true_flag": True,
+        "default_true_user_false_flag": False,
+        "default_true_user_true_flag": True,
+    }
+
+
+@pytest.fixture
+def config(flags) -> AdapterRequiredConfig:
+    raw_config = {
+        "credentials": CredentialsStub("test_database", "test_schema"),
+        "profile_name": "test_profile",
+        "target_name": "test_target",
+        "threads": 4,
+        "project_name": "test_project",
+        "query_comment": QueryComment(),
+        "cli_vars": {},
+        "target_path": "path/to/nowhere",
+        "log_cache_events": False,
+    }
+    return SimpleNamespace(**raw_config)
+
+
+@pytest.fixture
+def behavior_flags() -> List[BehaviorFlag]:
+    return [
+        {
+            "name": "default_false_user_false_flag",
+            "default": False,
+            "docs_url": "https://docs.com",
+        },
+        {
+            "name": "default_false_user_true_flag",
+            "default": False,
+            "description": "This is a false flag.",
+        },
+        {
+            "name": "default_false_user_skip_flag",
+            "default": False,
+            "description": "This is a true flag.",
+        },
+        {
+            "name": "default_true_user_false_flag",
+            "default": True,
+            "description": "This is fake news.",
+        },
+        {
+            "name": "default_true_user_true_flag",
+            "default": True,
+            "docs_url": "https://moar.docs.com",
+        },
+        {
+            "name": "default_true_user_skip_flag",
+            "default": True,
+            "description": "This is a true flag.",
+        },
+    ]
+
+
+def test_register_behavior_flags(adapter):
```

Comment:
This is the same test as in `./test_behavior_flags.py`, but with `flags` removed from the `config` fixture above, and swapping the expected values for the two cases where the user override changes the result.